pidgin/app.py

```diff
@@ -120,17 +124,45 @@ def request_metadata(object_id):
     Write a query and transmit it to send_query().
     """
     file_type = get_file_type(object_id)
+    query_txt = build_query(object_id, file_type, True)
+    response = send_query(query_txt)
+
+    # if the file has no core metadata, get the other metadata only
+    if not has_core_metadata(response, file_type):
+        simple_query_txt = build_query(object_id, file_type, False)
+        response = send_query(simple_query_txt)
+
+    return response
+
+
+def has_core_metadata(response, file_type):
+    """
+    Return True if a query response contains core metadata, False otherwise.
+    """
+    try:
+        # try to access the core_metadata
+        response['data'][file_type][0]['core_metadata_collections'][0]
+    except:
+        return False
+    return True
+
 
-    # get the metadata from the type of file and the object_id
-    query_txt = '''{{ {} (object_id: "{}") {{
-        core_metadata_collections {{
-            title description creator contributor coverage
-            language publisher rights source subject
-        }}
-        type file_name data_format file_size
-        project_id object_id updated_datetime }} }}
-        '''.format(file_type, object_id)
-    return send_query(query_txt)
+def build_query(object_id, file_type, get_core_metadata):
```

Comment:
Would it make any sense to give `get_core_metadata` a default value? It seems like defaulting to False would be reasonable, what do you think?